### PR TITLE
Added TDD yaml

### PR DESCRIPTION
**Pull request by rachitjain2706**
- Added TDD yaml files
- Added support for codecov

### CODE REVIEW

File: .github/workflows/tdd.yml
Comment (zhengtr) [inline .github/workflows/tdd.yml:1]: Do we need to run `pylint` to check the code quality here? Just like [this](https://github.com/nyu-devops/lab-flask-bdd/blob/24da7b81cd1778bd961dc00ab9790bf2006e6457/.github/workflows/tdd-tests.yml#L53).

**Comment by sahildeshpande25** [comment]
image

The build threw the above warning. I think this can be fixed if we upgrade the action versions to actions/checkout@v3 and codecov/codecov-action@v3. [Ways to use versioned actions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-versioned-actions)

Linking [checkout](https://github.com/actions/checkout/blob/main/CHANGELOG.md#v300) and [codecov](https://github.com/codecov/codecov-action/blob/main/CHANGELOG.md#300) changelogs.

File: .github/workflows/tdd.yml
Comment (rachitjain2706) [reply .github/workflows/tdd.yml:1]: Added this in the `yaml` and fixed linting errors.

**Comment by rachitjain2706** [comment]
Made the v3 changes

**Review by zhengtr — APPROVED** [review]
All tests passed and CI build passed. LGTM

**Review by sahildeshpande25 — APPROVED** [review]
LGTM

**Pull request merged**